### ISSUE TITLE
BeginEnd Rule(s) not popped

### ISSUE THREAD

**Issue by alexandrudima** [Microsoft/TypeScript-TmLanguage#248]
Moved from Microsoft/vscode#10677

---

From @pocesar

Minimal reproducible example:
```typescript
class C1 {
    m1 = 1;
    static s1() {
        return [() => (new this).m1];
    }
}
class C2 {
    m1 = 1;
    static s1() {
        return [() => (new this).m1];
    }
}
class C3 {
    m1 = 1;
    static s1() {
        return [() => (new this).m1];
    }
}
class C4 {
    m1 = 1;
    static s1() {
        return [() => (new this).m1];
    }
}
class C5 {
    m1 = 1;
    static s1() {
        return [() => (new this).m1];
    }
}
class C6 {
    m1 = 1;
    static s1() {
        return [() => (new this).m1];
    }
}
```

![image](https://cloud.githubusercontent.com/assets/5047891/17851062/73b5ea36-6860-11e6-9b3a-6e759fed9686.png)

**Comment by alexandrudima** [comment]
Grammar analysis on this file:

[out.txt](https://github.com/Microsoft/TypeScript-TmLanguage/files/429824/out.txt)

**Comment by sheetalkamat** [comment]
Fixed by #257
